### PR TITLE
Updated for the new version of PACMANFrame.hpp.

### PR DESCRIPTION
**Pull request by krisfur**
Updates to NDReadoutTypes.hpp and some others to properly handle the new PACMANFrame now included in dataformats as per issue #103

### CODE REVIEW

**Comment by dingp** [comment]
FWIW, my build with commit 1689afe failed with:

```
/build/dingpf/github-runner/actions-runner/_work/daq-release/daq-release/dev/sourcecode/readout/include/readout/models/DefaultRequestHandlerModel.hpp:366:77: error: ‘frames_per_element’ is not a member of ‘dunedaq::readout::types::PACMAN_MESSAGE_STRUCT’
                 element->get_timestamp() + (ReadoutType::frames_per_element - 1) * ReadoutType::tick_dist >=
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/build/dingpf/github-runner/actions-runner/_work/daq-release/daq-release/dev/sourcecode/readout/include/readout/models/DefaultRequestHandlerModel.hpp:366:82: error: ‘tick_dist’ is not a member of ‘dunedaq::readout::types::PACMAN_MESSAGE_STRUCT’
                 element->get_timestamp() + (ReadoutType::frames_per_element - 1) * ReadoutType::tick_dist >=
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
/build/dingpf/github-runner/actions-runner/_work/daq-release/daq-release/dev/sourcecode/readout/include/readout/models/DefaultRequestHandlerModel.hpp:369:47: error: ‘struct dunedaq::readout::types::PACMAN_MESSAGE_STRUCT’ has no member named ‘begin’
               for (auto frame_iter = element->begin(); frame_iter != element->end(); frame_iter++) {
                                      ~~~~~~~~~^~~~~
/build/dingpf/github-runner/actions-runner/_work/daq-release/daq-release/dev/sourcecode/readout/include/readout/models/DefaultRequestHandlerModel.hpp:369:79: error: ‘struct dunedaq::readout::types::PACMAN_MESSAGE_STRUCT’ has no member named ‘end’
               for (auto frame_iter = element->begin(); frame_iter != element->end(); frame_iter++) {
 
```

**Comment by krisfur** [comment]
Yeah there has been some changes in readout since we wrote this and waited for dataformats changes to be approved. Will try to get this fixed tomorrow.

**Comment by dingp** [comment]
Sounds good! Thanks! Though you may bear the blame of breaking develop and nightly build in the mean time. 🤣

**Comment by krisfur** [comment]
@dingp can you check against commit 7e47864 ? It should've fixed the issue you've quoted above.

**Comment by dingp** [comment]
@krisfur Yes, that fixed the issue. I would like the fix to be in `develop` branch before nightly build runs. Should I merge the PR?

**Comment by krisfur** [comment]
If all builds then sure go ahead, it shouldn't impact operation for anyone but ND readout stuff (I think), and if that has bugs we can PR later.

**Comment by roland-sipos** [comment]
Please don't merge before @jimbrooke or me gives a green light on PRs, or reviewers are assigned. I'm reverting this.

File: include/readout/models/DefaultRequestHandlerModel.hpp [hunk 1]
```diff
@@ -402,6 +402,7 @@ class DefaultRequestHandlerModel : public RequestHandlerConcept<ReadoutType, Lat
       // Requeue if needed
       if (rres.result_code == ResultCode::kNotYet) {
         if (m_run_marker.load()) {
+
```
Comment (roland-sipos) [inline include/readout/models/DefaultRequestHandlerModel.hpp:405]: Why the extra line? Just checkout the DefReqHandler.

File: include/readout/models/ReadoutModel.hpp [hunk 2]
```diff
@@ -242,6 +242,7 @@ class ReadoutModel : public ReadoutConcept
     while (m_run_marker.load()) {
       try {
         auto timesyncmsg = dfmessages::TimeSync(m_raw_processor_impl->get_last_daq_time());
+        //std::cout << "Get last DAQ Time: " << m_raw_processor_impl->get_last_daq_time() << std::endl;
```
Comment (roland-sipos) [inline include/readout/models/ReadoutModel.hpp:245]: Please remove.

File: include/readout/models/ReadoutModel.hpp [hunk 3]
```diff
@@ -252,9 +253,10 @@ class ReadoutModel : public ReadoutConcept
 
           if (m_fake_trigger) {
             dfmessages::DataRequest dr;
-            dr.trigger_timestamp = timesyncmsg.daq_time > 500 * us ? timesyncmsg.daq_time - 500 * us : 0;
-            auto width = 10000;
-            uint offset = 100;
+           
+            dr.trigger_timestamp = timesyncmsg.daq_time > 1 ? timesyncmsg.daq_time + 1 : 0;
```
Comment (roland-sipos) [inline include/readout/models/ReadoutModel.hpp:257]: This completely breaks the fake trigger functionality for WIBs and DAPHNE. PACMAN should move to nanoseconds, and this fake trigger should be configurable. But don't overwrite defaults, as it breaks stuff runtime for emulators.

**Review by roland-sipos — COMMENTED** [review]
Breaks runtime behavior.

**Comment by roland-sipos** [comment]
I'm not sure if the linter was run on this. If not, please do so.

**Comment by dingp** [comment]
> @dingp please don't merge before @jimbrooke or me gives a green light on PRs, or reviewers are assigned. I'm reverting this.

Whoops, clicked the button too soon after I saw Krzysztof's comment. Sorry for the inconvenience.

**Comment by willpvazquez** [comment]
FYI - the underlying issues should now be fixed in https://github.com/DUNE-DAQ/readout/pull/106

**Pull request merged**